test/conf_include_test.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2016-2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <openssl/conf.h>
+#include <openssl/err.h>
+#include <openssl/ssl.h>
+#include <string.h>
+#include "testutil.h"
+
+#ifdef _WIN32
+# include <direct.h>
+# define DIRSEP '\\'
+# define CHDIR _chdir
+#else
+# include <unistd.h>
+# define DIRSEP '/'
+# define CHDIR chdir
+#endif
+
+/* changes path to that of the filename */
+static int change_path(const char *file)
+{
+    char *s = OPENSSL_strdup(file);
+    char *p;
+    int ret;
+
+    if (s == NULL)
+        return -1;
+
+    p = strrchr(s, DIRSEP);
+    if (p == NULL)
+        return 0;
+    *p = 0;
+
+    TEST_note("changing path to %s", s);
+    ret = CHDIR(s);
```

Comment:
On related note I'm not exactly fond of CHDIR. I'd rather see same pattern as with stat, i.e. chdir being defined as _chdir. [Just in case for reference, having those redefinitions serve different purposes in library and applications. Well, rather that in library they are must, while in applications it's just about shutting up compiler warnings. I mean one could have chosen to suppress them in applications, but as long as one uses same flags it would effectively prevent static libcrypto to be encapsulated in another dll.]